subprojects/mockito-scala/src/main/scala/org/mockito/MockitoSugar.scala

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright © 2017 Morgan Stanley.  All rights reserved.
+ *
+ * THIS SOFTWARE IS SUBJECT TO THE TERMS OF THE MIT LICENSE.
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+ *
+ * IN ADDITION, THE FOLLOWING DISCLAIMER APPLIES IN CONNECTION WITH THIS SOFTWARE:
+ * THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.
+ */
+
+package org.mockito
+
+import org.mockito.MockitoEnhancerUtil.stubMock
+import org.mockito.MockitoSugar.clazz
+import org.mockito.stubbing.{Answer, OngoingStubbing, Stubber}
+import org.mockito.verification.{VerificationMode, VerificationWithTimeout}
+
+import scala.collection.JavaConverters._
+import scala.reflect.ClassTag
+
+private[mockito] trait DoSomething {
+    /**
+      * Delegates the call to <code>Mockito.doReturn(toBeReturned, toBeReturnedNext)</code>
+      * but fixes the following compiler issue that happens because the overloaded vararg on the Java side
+      *
+      * Error:(33, 25) ambiguous reference to overloaded definition,
+      * both method doReturn in class Mockito of type (x$1: Any, x$2: Object*)org.mockito.stubbing.Stubber
+      * and  method doReturn in class Mockito of type (x$1: Any)org.mockito.stubbing.Stubber
+      * match argument types (`Type`)
+      *
+      */
+    def doReturn[T](toBeReturned: T, toBeReturnedNext: T*): Stubber =
+        Mockito.doReturn(toBeReturned, toBeReturnedNext.map(_.asInstanceOf[Object]): _*)
```

Comment:
Check if it works by default without mapping to Object

Reply:
Sadly, it doesn't